src/main/scala/lms/core/stub.scala

```diff
@@ -1388,19 +1388,22 @@ abstract class DslDriverCPP[A: Manifest, B: Manifest] extends DslDriverC[A, B] w
 abstract class CompilerC[A:Manifest, B:Manifest] extends DslDriverC[A, B] { q =>
 
   // get original graph
-  val graph = Adapter.genGraph1(manifest[A], manifest[B])(x => Unwrap(wrapper(Wrap[A](x))))
+  val initial_graph = Adapter.genGraph1(manifest[A], manifest[B])(x => Unwrap(wrapper(Wrap[A](x))))
 
   // run some transformation
-  def transform(graph: Graph): Graph = graph
+  def transform(graph: Graph): (List[String], Graph) = (List(graph.toString), graph)
```

Comment:
Does it make sense to return a `List[Graph]` which contains all intermediate graphs instead of `(List[String], Graph)`? And the first/last one in the list can be the final/initial graph. I just felt it is not `transform`'s job to return several string representations of the graphs.

Reply:
That can be done